src/backend/distributed/utils/maintenanced.c

```diff
@@ -272,11 +285,10 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		{
 			StartTransactionCommand();
 
-			/*
-			 * We don't want to run the deadlock checks if there exists
-			 * any version mistmatch.
-			 */
-			if (CheckCitusVersion(DEBUG1))
+			/* to avoid the extension being dropped while we're using it */
+
+			if (LockCitusExtension() && CheckCitusVersion(DEBUG1) &&
```

Comment:
A debug mesage around here saying that we're skipping deadlock detection would be helpful.